### PR TITLE
Add bundles and charms from profile

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- open your profile (using the `profile` flag)
- toggle open the details for a bundle or charm
- click the add to model button
- the profile should close and the bundle/charm should get added to the model

**Comment by huwshimi** [comment]
[Allowing adding of profile charms/bundles to the current model](https://trello.com/c/litRjP4u/133-allowing-adding-of-profile-charms-bundles-to-the-current-model)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8363/

File: jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js [hunk 1]
```diff
@@ -177,6 +180,7 @@ class ProfileBundleList extends React.Component {
 };
 
 ProfileBundleList.propTypes = {
+  acl: PropTypes.object.isRequired,
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js]: Please use shapeup when requiring a shape like acl.

File: jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js [hunk 2]
```diff
@@ -185,6 +189,8 @@ ProfileBundleList.propTypes = {
     list: PropTypes.func.isRequired,
     url: PropTypes.string.isRequired
   }).isRequired,
+  deployTarget: PropTypes.func.isRequired,
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js]: could those two be grouped in a shape,like modelAPI?

Comment (frankban) [reply jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js]: Also should deployTarget be renamed to addToCanvas or similar? deployTarget seems a bit ambiguous, what do you think?

Comment (huwshimi) [reply jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js]: Good point, that method is used all over the place so I've filed #3388 for a followup.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8368/

**Comment by huwshimi** [comment]
Thanks @frankban, that bug is fixed now. :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**